frontend/server/src/main/java/org/pytorch/serve/grpcimpl/ManagementImpl.java

```diff
@@ -0,0 +1,193 @@
+                            null);
+            sendStatusResponse(responseObserver, statusResponse);
+        } catch (ExecutionException | InterruptedException e) {
+            sendException(responseObserver, e, "Error while creating workers");
+        } catch (ModelNotFoundException | ModelVersionNotFoundException e) {
+            sendErrorResponse(responseObserver, Status.NOT_FOUND, e);
+        } catch (BadRequestException e) {
+            sendErrorResponse(responseObserver, Status.INVALID_ARGUMENT, e);
+        }
+    }
+
+    @Override
+    public void setDefault(
+            SetDefaultRequest request, StreamObserver<ManagementResponse> responseObserver) {
+        String modelName = request.getModelName();
+        String newModelVersion = request.getModelVersion();
+
+        try {
+            String msg = ApiUtils.setDefault(modelName, newModelVersion);
+            sendResponse(responseObserver, msg);
+        } catch (ModelNotFoundException | ModelVersionNotFoundException e) {
+            sendErrorResponse(responseObserver, Status.NOT_FOUND, e);
+        }
+    }
+
+    @Override
+    public void unregisterModel(
+            UnregisterModelRequest request, StreamObserver<ManagementResponse> responseObserver) {
+        try {
+            String modelName = request.getModelName();
+            if (modelName == null || ("").equals(modelName)) {
+                sendErrorResponse(
+                        responseObserver,
+                        Status.INVALID_ARGUMENT,
+                        new BadRequestException("Parameter url is required."));
+            }
+
+            String modelVersion = request.getModelVersion();
+
+            if (("").equals(modelVersion)) {
+                modelVersion = null;
+            }
+            ApiUtils.unregisterModel(modelName, modelVersion);
+            String msg = "Model \"" + modelName + "\" unregistered";
+            sendResponse(responseObserver, msg);
+        } catch (ModelNotFoundException | ModelVersionNotFoundException e) {
+            sendErrorResponse(responseObserver, Status.NOT_FOUND, e);
+        } catch (BadRequestException e) {
+            sendErrorResponse(responseObserver, Status.INVALID_ARGUMENT, e);
+        }
+    }
+
+    private void sendResponse(StreamObserver<ManagementResponse> responseObserver, String msg) {
+        ManagementResponse reply = ManagementResponse.newBuilder().setMsg(msg).build();
+        responseObserver.onNext(reply);
+        responseObserver.onCompleted();
+    }
+
+    private void sendErrorResponse(
+            StreamObserver<ManagementResponse> responseObserver,
+            Status status,
+            String description,
+            String errorClass) {
+        responseObserver.onError(
+                status.withDescription(description)
+                        .augmentDescription(errorClass)
+                        .asRuntimeException());
+    }
+
+    private void sendErrorResponse(
```

Comment:
@maaquib: Both Inference APIs and Management APIs use different responseObserver types. Also, the inference API has only one main API (predict). Thus for avoiding one minor code duplication we will need to add a bunch of complicated type castings.

I have refactored the InferenceImpl.java to avoid code duplication within the file for sending error response.